src/test/app/Flow_test.cpp

```diff
@@ -1392,9 +1396,16 @@ struct Flow_test : public beast::unit_test::suite
         auto const bob = Account("bob");
         auto const gw = Account("gw");
         auto const USD = gw["USD"];
+        auto const usdC = USD.currency;
 
         env.fund(XRP(10000), alice, bob, gw);
```

Comment:
So the problem was that the `fset()` in `Env::fund()`  slipped in front of the upcoming call to `trust()`?

I'm wondering why we always see the `pay()` and `fset()` in `Env::fund` as well ordered.  That certainly seems to be the case, otherwise the `fset()` would fail regularly.  But I'm unsure why that would be the case.

Still, we could potentially save others the debugging pain you went through by appending a call to `close()` to the end of `Env::fund()`.  That may be a (much) bigger change than you had in mind.  I'll just offer it as a thought.